src/traces/image/defaults.js

```diff
@@ -26,10 +32,16 @@ module.exports = function supplyDefaults(traceIn, traceOut) {
     coerce('y0');
     coerce('dx');
     coerce('dy');
-    var colormodel = coerce('colormodel');
 
-    coerce('zmin', constants.colormodel[colormodel].min);
-    coerce('zmax', constants.colormodel[colormodel].max);
+    if(traceOut._isFromZ) {
+        coerce('colormodel');
+        coerce('zmin', constants.colormodel[traceOut.colormodel].min);
+        coerce('zmax', constants.colormodel[traceOut.colormodel].max);
+    } else if(traceOut._isFromSource) {
+        traceOut.colormodel = 'rgba';
```

Comment:
Hmm...
We need to update description for `colormodel` https://github.com/plotly/plotly.js/blob/3781f482455e7623d1a0368b6e74427df549550b/src/traces/image/attributes.js#L33-L40
Or even better if you not set this parameter this way.
What about adding an underscore variable e.g.
```js
traceOut._colormodel = 'rgba';
```
here and then 
```js
var colormodel = trace._colormodel || trace.colormodel;
```
in `calc`?

Reply:
Wondering also what happens if one pass a `jpeg` (often has no alpha channel) instead of a `png`? 
In that case the default should be `rgba` or simply `rgb`?

Reply:
Codepen for `jpeg`: https://codepen.io/MojtabaSamimi/pen/zYqKOKy

This seems correct to me.

cc @alexcjohnson

Reply:
Thinking more about this `automatic` default behaviour.
I still believe we should respect `colormodel` to be defined by the user.
And we must remove `colormodel.dflt` and set it during `supplyDefaults`.
```js
if(traceOut._isFromZ) {
  coerce('colormodel', 'rgb');
  ...
} else if(traceOut._isFromSource) {
  coerce('colormodel', isJPEG(traceOut.source) ? 'rgb' : 'rgba');
  ...
}
```

Reply:
> I still believe we should respect colormodel to be defined by the user.

Do you mean supporting colormodel `hsl` with say a PNG image? That seems like a weird thing to do and support. Also, it would be slow: now we would have to read all the pixels one by one and convert them into the new colorspace. cc @archmoj

Reply:
cc @alexcjohnson @nicolaskruchten :point_up: